src/Components/Aspire.Seq/AspireSeqExtensions.cs

```diff
@@ -23,47 +24,41 @@ public static class AspireSeqExtensions
-    public static void AddSeqEndpoint(this IHostApplicationBuilder builder, string connectionName, Action<SeqSettings>? configureSettings = null)
+    public static void AddSeqEndpoint(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<SeqSettings>? configureSettings = null)
     {
         ArgumentNullException.ThrowIfNull(builder);
 
         var settings = new SeqSettings();
+        settings.ServerUrl = builder.Configuration.GetConnectionString(connectionName);
+        settings.Logs.Protocol = OtlpExportProtocol.HttpProtobuf;
+        settings.Traces.Protocol = OtlpExportProtocol.HttpProtobuf;
+
         builder.Configuration.GetSection("Aspire:Seq").Bind(settings);
+        configureSettings?.Invoke(settings);
 
-        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        if (!string.IsNullOrEmpty(settings.ServerUrl))
         {
-            settings.ServerUrl = connectionString;
+            settings.Logs.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/logs");
+            settings.Traces.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/traces");
         }
-
-        configureSettings?.Invoke(settings);
-
-        if (string.IsNullOrEmpty(settings.ServerUrl))
+        if (!string.IsNullOrEmpty(settings.ApiKey))
         {
-            settings.ServerUrl = "http://localhost:5341";
+            settings.Logs.Headers = string.IsNullOrEmpty(settings.Logs.Headers) ? $"X-Seq-ApiKey={settings.ApiKey}" : $"{settings.Logs.Headers},X-Seq-ApiKey={settings.ApiKey}";
+            settings.Traces.Headers = string.IsNullOrEmpty(settings.Traces.Headers) ? $"X-Seq-ApiKey={settings.ApiKey}" : $"{settings.Traces.Headers},X-Seq-ApiKey={settings.ApiKey}";
         }
 
-        builder.Services.Configure<OpenTelemetryLoggerOptions>(logging => logging.AddOtlpExporter(opt =>
-        {
-            opt.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/logs");
-            opt.Protocol = OtlpExportProtocol.HttpProtobuf;
-            if (!string.IsNullOrEmpty(settings.ApiKey))
-            {
-                opt.Headers = $"X-Seq-ApiKey={settings.ApiKey}";
-            }
-        }));
-        builder.Services.ConfigureOpenTelemetryTracerProvider(tracing => tracing
-            .AddOtlpExporter(opt =>
-                {
-                    opt.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/traces");
-                    opt.Protocol = OtlpExportProtocol.HttpProtobuf;
-                    if (!string.IsNullOrEmpty(settings.ApiKey))
-                    {
-                        opt.Headers = $"X-Seq-ApiKey={settings.ApiKey}";
-                    }
-                }
+        builder.Services.Configure<OpenTelemetryLoggerOptions>(logging => logging.AddProcessor(
+            sp => new BatchLogRecordExportProcessor(new OtlpLogExporter(settings.Logs))
```

Comment:
@liammclennan If you want to respect `ExportProcessorType` make this...

```csharp
        builder.Services.Configure<OpenTelemetryLoggerOptions>(logging => logging.AddProcessor(
            sp => settings.Logs.ExportProcessorType == ExportProcessorType.Batch
                ? new BatchLogRecordExportProcessor(new OtlpLogExporter(settings.Logs))
                : new SimpleLogRecordExportProcessor(new OtlpLogExporter(settings.Logs))
            ));
```

Similar for below in tracing.

Reply:
Thanks for the review. I've made this change.